content/en/docs/concepts/overview/kubernetes-api.md

```diff
@@ -22,63 +22,131 @@ external components communicate with one another.
-        <td><code>gzip</code></td>
-        <td><em>not supplying this header is also acceptable</em></td>
-     </tr>
-     <tr>
-        <td rowspan="3"><code>Accept</code></td>
-        <td><code>application/com.github.proto-openapi.spec.v2@v1.0+protobuf</code></td>
-        <td><em>mainly for intra-cluster use</em></td>
-     </tr>
-     <tr>
-        <td><code>application/json</code></td>
-        <td><em>default</em></td>
-     </tr>
-     <tr>
-        <td><code>*</code></td>
-        <td><em>serves </em><code>application/json</code></td>
-     </tr>
-  </tbody>
-</table>
+## Unaggregated Discovery
 
-Kubernetes implements an alternative Protobuf based serialization format that
-is primarily intended for intra-cluster communication. For more information
-about this format, see the [Kubernetes Protobuf serialization](https://git.k8s.io/design-proposals-archive/api-machinery/protobuf.md) design proposal and the
-Interface Definition Language (IDL) files for each schema located in the Go
-packages that define the API objects.
+Without discovery aggregation, discovery is published in levels, with the root
+endpoints publishing discovery information for downstream documents.
+
+A list of all group versions supported by a cluster is published at
+the `/api` and `/apis` endpoints. Example:
+
+```
+{
+  "kind": "APIGroupList",
+  "apiVersion": "v1",
+  "groups": [
+    {
+      "name": "apiregistration.k8s.io",
+      "versions": [
+        {
+          "groupVersion": "apiregistration.k8s.io/v1",
+          "version": "v1"
+        }
+      ],
+      "preferredVersion": {
+        "groupVersion": "apiregistration.k8s.io/v1",
+        "version": "v1"
+      }
+    },
+    {
+      "name": "apps",
+      "versions": [
+        {
+          "groupVersion": "apps/v1",
+          "version": "v1"
+        }
+      ],
+      "preferredVersion": {
+        "groupVersion": "apps/v1",
+        "version": "v1"
+      }
+    },
+    ...
+}
+```
+
+Additional requests are needed to obtain the discovery document for each group version at
+`/apis/<group>/<version>` (for example:
+`/apis/rbac.authorization.k8s.io/v1alpha1`), which advertises the list of
+resources served under a particular group version. These endpoints are used by
+kubectl to fetch the list of resources supported by a cluster.
+
+<!-- body -->
+
+## OpenAPI specification {#api-specification}
+
+For details on the specification of the OpenAPI format, refer to the [OpenAPI documentation](https://www.openapis.org/).
+
+Kubernetes serves both OpenAPI V2.0 and OpenAPI V3.0. OpenAPI V3 is the
+preferred method of accessing the OpenAPI because it offers a more comprehensive
+(lossless) representation of Kubernetes resources. Due to limitations of OpenAPI
+V2, certain fields are dropped from the published OpenAPI including but not
+limited to `default`, `nullable`, `oneOf`. At the moment, Kubernetes publishes
+OpenAPI V3.0, and there are no plans to support 3.1 in the near future.
```

Comment:
We [avoid statements about the future](/docs/contribute/style/style-guide/#avoid-statements-about-the-future).
I'd try this
```suggestion
limited to `default`, `nullable`, `oneOf`. Kubernetes {{< skew currentVersion >}}
publishes OpenAPI v2.0 and v3.0; there are no plans to support 3.1 in the near future.
```

We might want to move this statement under the v3 heading, rather than here.